test/test_irods_http_api.py

```diff
@@ -1107,6 +1107,111 @@ def test_support_for_genquery1(self):
+
+        resc_name = 'test_ufs_genquery1_no_distinct'
+
+        try:
+            # Create a unixfilesystem resource.
+            r = requests.post(f'{self.url_base}/resources', headers=rodsadmin_headers, data={
+                'op': 'create',
+                'name': resc_name,
+                'type': 'unixfilesystem',
+                'host': self.server_hostname,
+                'vault-path': os.path.join('/tmp', f'{resc_name}_vault')
+            })
+            #print(r.content) # Debug
+            self.assertEqual(r.status_code, 200)
+            self.assertEqual(r.json()['irods_response']['status_code'], 0)
+
+            # Replicate the data object.
+            r = requests.post(f'{self.url_base}/data-objects', headers=rodsuser_headers, data={
+                'op': 'replicate',
+                'lpath': data_object,
+                'dst-resource': resc_name
+            })
+            #print(r.content) # Debug
+            self.assertEqual(r.status_code, 200)
+            self.assertEqual(r.json()['irods_response']['status_code'], 0)
+
+            coll_name = os.path.dirname(data_object)
+            data_name = os.path.basename(data_object)
+            r = requests.get(self.url_endpoint, headers=rodsuser_headers, params={
+                'op': 'execute_genquery',
+                'parser': 'genquery1',
+                'distinct': 0,
+                'query': f"select DATA_NAME where COLL_NAME = '{coll_name}' and DATA_NAME = '{data_name}'"
+            })
+            #print(r.content) # Debug
+            self.assertEqual(r.status_code, 200)
+            result = r.json()
+            self.assertEqual(result['irods_response']['status_code'], 0)
+            self.assertEqual(len(result['rows']), 2)
+
+        finally:
+            # Remove data object.
+            requests.post(f'{self.url_base}/data-objects', headers=rodsuser_headers, data={
+                'op': 'remove',
+                'lpath': data_object,
+                'no-trash': 1
+            })
+
+            # Remove resource.
+            requests.post(f'{self.url_base}/resources', headers=rodsadmin_headers, data={
+                'op': 'remove',
+                'name': resc_name
+            })
+
+    def test_genquery1_case_sensitivity_option(self):
+        headers = {'Authorization': 'Bearer ' + self.rodsuser_bearer_token}
+
+        # Create a data object with mixed case letters in the name.
+        data_object = os.path.join('/', self.zone_name, 'home', self.rodsuser_username, 'test_GenQuery1_NO_DISTINCT.TxT')
+        r = requests.post(f'{self.url_base}/data-objects', headers=headers, data={'op': 'touch', 'lpath': data_object})
+        #print(r.content) # Debug
+        self.assertEqual(r.status_code, 200)
+        result = r.json()
+        self.assertEqual(result['irods_response']['status_code'], 0)
+
+        coll_name = os.path.dirname(data_object)
+        data_name = os.path.basename(data_object)
```

Comment:
do you need a `lower()` here or something to make sure you're searching for something different than is in the catalog?

Reply:
Done.